src/inversion_scripts/utils.py

```diff
@@ -410,3 +413,53 @@ def get_strdate(current_time, date_threshold):
         strdate = current_time.floor("60min").strftime("%Y%m%d_%H")
  
     return strdate
+
+
+def filter_prior_files(filenames, start_date, end_date):
+    """
+    Filter a list of HEMCO diagnostic files based on the specified date range.
+    """
+    # Parse the input dates
+    start_date = datetime.strptime(start_date, "%Y%m%d")
+    end_date = datetime.strptime(end_date, "%Y%m%d") - timedelta(days=1)
+
+    filtered_files = []
+    for file in filenames:
+        # Extract the date part from the filename
+        date_str = file.split('.')[1]
+        file_date = datetime.strptime(date_str, "%Y%m%d%H%M")
+
+        # Check if the file date is within the specified range
+        if start_date <= file_date <= end_date:
+            # Create a key for the year and month
+            year_month = file_date.strftime("%Y%m")
```

Comment:
I think this is leftover.

Reply:
Oops, yup. Cleaned that up and a misleading comment from previous implementation